# Medical Insurance Project
### Take the dataset and do something with it
The dataset has these variables:
* age: in years (integer)
* sex: male/female (string)
* bmi: body mass index (float)
* children: number of children (integer)
* smoker: yes/no (string)
* region: southwest, southeast, northwest or northeast (string)
* charges: cost of insurance (float)

Ideas: 
* bmi and age vs. charges (which results in higher rates: being old or being obese)
    * split into four groups: old and obese, old and not obese, young and obese, young and not obese
* which region has the highest bmi

### Comparing how age and obesity affects insurance cost
My hypothesis: young/not obese would result in the lowest cost and old/obese would result in the highest cost; I think that obesity will affect the cost more than age

In [116]:
#### import csv

#function to find averages
def make_average(total, count):
        average = total / count
        return average

#Finding the average age to see what classifies as 'young' and 'old'    
with open('insurance.csv') as insurance_data:
    reader = csv.DictReader(insurance_data)
    total_age = 0
    age_count= 0
    for row in reader: 
        total_age += int(row['age'])
        age_count += 1
    print(total_age, age_count)
    
    average_age = make_average(total_age, age_count)
    print("The average age of the data set is", average_age)
    
with open('insurance.csv') as insurance_data:
    reader = csv.DictReader(insurance_data)
    
    # Split data into 4 groups: 
    ## old_ob (old and obese), old_no_ob (old and not obese), yo_ob (young and obese), yo_no_ob (young and not obese)
    def split(data):
        old_ob_total = 0
        old_ob_count = 0
        old_no_ob_total = 0
        old_no_ob_count = 0
        yo_ob_total = 0
        yo_ob_count = 0
        yo_no_ob_total = 0
        yo_no_ob_count = 0
        for row in reader:
            if float(row['age']) > 39 and float(row['bmi']) >= 30:
                old_ob_total += float(row['charges'])
                old_ob_count += 1
            elif float(row['age']) > 39 and float(row['bmi']) < 30:
                old_no_ob_total += float(row['charges'])
                old_no_ob_count += 1
            elif float(row['age']) <= 39 and float(row['bmi']) >= 30:
                yo_ob_total += float(row['charges'])
                yo_ob_count += 1
            else:
                yo_no_ob_total += float(row['charges'])
                yo_no_ob_count += 1
        
        #finding the cost for each of the groups
        old_ob_avg = make_average(old_ob_total, old_ob_count)
        old_no_ob_avg = make_average(old_no_ob_total, old_no_ob_count)
        yo_ob_avg = make_average(yo_ob_total, yo_ob_count)
        yo_no_ob_avg = make_average(yo_no_ob_total, yo_no_ob_count)
        
        return print("The average cost for old/obese is: " + str(old_ob_avg) + " \n" + 
                     "The average cost for old/not obese is: " + str(old_no_ob_avg) + "\n" +
                     "The average cost for young/obese is: " + str(yo_ob_avg) + "\n" +
                     "The average cost for young/not obese is: " + str(yo_no_ob_avg))
    split(reader)


52459 1338
The average age of the data set is 39.20702541106129
The average cost for old/obese is: 18069.23041601583 
The average cost for old/not obese is: 14251.305311368418
The average cost for young/obese is: 12644.094051280488
The average cost for young/not obese is: 7799.716186499998


### Results
* being old (> 39 years) and obese (bmi >= 30) results in the highest average insurance cost (~18,069 dollars)
* being young (<= 39) and not obese (bmi < 30) results in the lowest average insurance cost (~7,800 dollars)

Obesity increases cost by ~3,818 dollars for the old group and ~4,844 dollars for the young group.
Age increases cost by ~5,425 dollars for the obese group and ~6,452 dollars for the not obese group.

Therefore, age increases insurance costs more than obesity.